jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/FloydWarshallShortestPaths.java

```diff
@@ -298,16 +310,6 @@ private void lazyCalculateMatrix()
                 }
```

Comment:
@d-michail based on git history, it looks like the Double.compares were introduced by you last October, so maybe you can answer?  (According to java.lang.Double javadoc, the difference has to do with NaN's and signed zeros.)